AUTH-README.md

```diff
@@ -0,0 +1,59 @@
+# Authentication
+
+As outlined <a href="https://github.com/grpc/grpc-common/blob/master/grpc-auth-support.md">here</a> gRPC supports a number of different mechanisms for asserting identity between an client and server. We'll present some code-samples here demonstrating how to provide SSL/TLS support encryption and identity assertions as well as passing OAuth2 tokens to services that support it.
+
+# Java 7, HTTP2 & Crypto
+
+## Cipher-Suites
+Java 7 does not support the <a href="https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-9.2.2">the cipher suites recommended</a> by the HTTP2 specification. To address this we suggest servers use Java 8 where possible. If this is not possible it is possible to use other ciphers but you need to ensure that the services you intend to call have <a href="https://github.com/grpc/grpc/issues/681">allowed out-of-spec ciphers</a>. Android does not suffer from this issue as it uses OpenSSL instead of SSLEngine to provide these features.
+
+## Protocol Negotiation (TLS-ALPN)
+HTTP2 mandates the use of <a href="https://tools.ietf.org/html/draft-ietf-tls-applayerprotoneg-05">ALPN</a> to negotiate the use of the protocol over SSL. No standard Java release has built-in support for ALPN today (<a href="https://bugs.openjdk.java.net/browse/JDK-8051498">there is a tracking issue</a> so go upvote it!) so we need to use the <a href="https://github.com/jetty-project/jetty-alpn">Jetty-ALPN</a> bootclasspath extension for OpenJDK to make it work.
+
+```sh
+java -Xbootclasspath/p:/path/to/jetty/alpn/extension.jar ...
+```
+
+Note that you must use the release of the Jetty-ALPN jar specific to the version of Java you are using.
+
+An option is provided to use GRPC over plaintext without TLS which is convenient for testing environments but users must be aware of the secuirty risks of doing so for real production systems.
```

Comment:
Argh, left the comma out. :) "An option is provided to use GRPC over plaintext without TLS. This is convenient for testing environments, however users must be aware of the security risks of doing so for real production systems."